### ISSUE TITLE
Problem when combining tile maps with normal plots

### ISSUE THREAD

**Issue by bkueng** [bokeh/bokeh#4860]
I try to add a tile map and figures to a column layout, but the map does not show
properly (stays blank) and I get this message in the Browser console:

```
Bokeh: could not set initial ranges
```

The problem can be reproduced with the following code (adaptation of
`tile_source.py` example):

```
from __future__ import print_function

from bokeh.util.browser import view
from bokeh.document import Document
from bokeh.embed import file_html
from bokeh.resources import INLINE

from bokeh.plotting import output_file
from bokeh.models import Plot
from bokeh.models import Range1d
from bokeh.models import WheelZoomTool, PanTool, BoxZoomTool
from bokeh.models import WMTSTileSource
from bokeh.layouts import column
from bokeh.plotting import figure


output_file("tile_source_example.html", title="Tile Source Example")

plots = []

plots_width = 800
plots_height = 400

# set to roughly full extent of web mercator projection
x_range = Range1d(start=-20000000, end=20000000)
y_range = Range1d(start=-20000000, end=20000000)

# create tile source from templated url
tile_options = {}
tile_options['url'] = 'http://c.tile.openstreetmap.org/{z}/{x}/{y}.png'
tile_source = WMTSTileSource(**tile_options)

# instantiate plot and add tile source
p = figure(x_range=x_range, y_range=y_range)
p.plot_width = plots_width
p.plot_height = plots_height

tile_renderer_options = {}
p.add_tile(tile_source, **tile_renderer_options) # works when commenting this

plots.append(p)

x = list(range(11))
y0 = x
y1 = [10-i for i in x]
y2 = [abs(i-5) for i in x]

# create a new plot
s1 = figure(title=None)
s1.circle(x, y0, size=10, color="navy", alpha=0.5)
s1.plot_width = plots_width
s1.plot_height = plots_height

plots.append(s1)

# create a new plot
s1 = figure(title=None)
s1.circle(x, y0, size=10, color="navy", alpha=0.5)
s1.plot_width = plots_width
s1.plot_height = plots_height

plots.append(s1) # also works when commenting this out


doc = Document()
doc.add_root(column(plots))

if __name__ == "__main__":
    filename = "tile_source.html"
    with open(filename, "w") as f:
        f.write(file_html(doc, INLINE, "Tile Source Example"))
    print("Wrote %s" % filename)
    view(filename)
```

Commenting either of the two marked lines makes it work.
Am I doing something wrong?

bokeh info:

```
Python version      :  3.5.1 (default, Jul 10 2016, 20:36:01)
IPython version     :  5.0.0
Bokeh version       :  0.12.0
BokehJS static path :  /opt/python_virtual_env/bokeh3/lib/python3.5/site-packages/bokeh/server/static
```

**Comment by aeroevan** [comment]
I am seeing the same issue. It looks like it's an issue with the x/y ranges inside the layout?
```[bokeh] could not set initial ranges```
is in my console when stacking figures (one with a tile source, one without).

I've found that including the reset tool, and resetting the figure with the tile source (only once or it isn't square...) seems to work, but is annoying.

**Comment by aeroevan** [comment]
Should mention that putting the img_tile figure in it's own column (`row([img_tile_figure, columns])`) works as expected.

**Comment by nanodan** [comment]
I'm having the same issue (reset once also corrects the issue). Do you know of a workaround I could fork and apply?

**Comment by bryevdv** [comment]
The original code is working for me with `0.12.10` (see screenshot) so I am closing this. @nanodan if you are seeing problems with `0.12.10` please open a new issue with your code to reproduce.

<img width="783" alt="screen shot 2017-10-20 at 14 28 48" src="https://user-images.githubusercontent.com/1078448/31838627-3db497a2-b5a3-11e7-9e26-4fb34c196fe4.png">
